metals/src/main/scala/scala/meta/internal/metals/formatting/MultilineString.scala

```diff
@@ -232,6 +266,16 @@ case class MultilineString(userConfig: () => UserConfiguration)
     }
   }
 
+  private def wasTripleQuoted(
+      splitLines: Array[String],
+      position: Position,
+  ): Boolean = {
+    val currentLine = splitLines(position.getLine - 1)
+    getIndexOfLastTripleQuote(currentLine).exists { case (_, quoteClosed) =>
+      !quoteClosed
+    }
```

Comment:
```suggestion
    val prevLine = splitLines(position.getLine - 1)
    val currLine = splitLines(position.getLine)
    prevLine.endsWith("\"\"\"") && !currLine.startsWith("\"\"\"")
```

Can we simplify it like this?
The idea is, that what we need to know are:
- Did the user enter a newline after `"""`?
- Is the triple quote closed (that means the current line of cursor starts with `"""`)? (maybe there's some whitespace for indent)

Reply:
Thank you so much for reviewing this Rikito :)

Well, frankly this was copied from `doubleQuoteNotClosed` whose logic was previously tested in the code.

As an example of the issue's trickiness, the `getIndexOfLastTripleQuote`, which was copied from `getIndexOfLastQuote` does not just get the index of last triple quote, but also It gets the index of the the last triple quote, which was not closed. 

So basically this prevents us from putting an extra triple quote if the previous line had two triple quotes: one begin, and one close.

Another tricky matter is that, this code action does itself generate a text edit to put the closing triple quotes in the `curLine`which is the line after the newly inserted newline character. 

So there won't be any triple quotes in it.

Reply:
Nevermind, @vzmerr pointed me out that it doesn't work for `"""foo|"""` and enter newline.

Reply:
I just noticed that the suggestions might lead to unexpected results in a few cases, for example
`val a = """" this is an open triple quoted string not ending with triple quotes`

Also, the function `getIndexOfLastTripleQuote` does more than just escaping the escaped ones. It also keeps opening and closing the triple quotes of the previous line, as it traverses through the characters to make sure that the last triple quote in the line is still open.
`a = """ first one """ + """ second one """ + """ third one which is not closed`

Also in case of `sql"""` the closing triple quotes do not get inserted automatically, so they are not present in the first place.